### ISSUE TITLE
ML performance and optimisation

### ISSUE THREAD

**Issue by ma595** [CALIPSO-project/SPINacc#69]
Optimise training data / ML bias (or better RERUN time)
- Remove LAI and NPP from predictors (from pre-run)
  - If re-run extension less than duration of pre-run -> WIN
- Vary number of Ncc:
  - Model dependent duration of site-runs ( duration known )
  - Total time of site-runs; Ncc x duration
How to optimise tradeoff rerun time vs number of pixel for site runs?

**Comment by ma595** [comment]
break this down into subtasks and prioritise. 

Without LAI and NPP performance was found to be poor. But they are expensive to obtain.

- provide an option to remove these from training.

**Comment by ma595** [comment]
Thanks @dsgoll123, I realise that (3) is not high priority but would you be able to provide these additional variables as soon as you can so that we are ready to use them when we have some time? I'll create individual issues so we can put them on the roadmap.

**Comment by dsgoll123** [comment]
> Thanks @dsgoll123, I realise that (3) is not high priority but would you be able to provide these additional variables as soon as you can so that we are ready to use them when we have some time? I'll create individual issues so we can put them on the roadmap.

Yes, we can use the other variables in that file for a test. It is a mixed bag which should contain useful variables but also some without much useful information. Can you work with this or do you want me to pre-select variables?

**Comment by dsgoll123** [comment]
> @ma595 asked why Ncc scales linearly with the computational demand of ORCHIDEE. If each pixel can be treated independently, why is this not embarrassingly parallel?

we are working on it to get there & should be there soon

**Comment by ma595** [comment]
![image](https://github.com/user-attachments/assets/83bf55cd-00a6-4caa-a88b-d6b33ab47019)

**Comment by ma595** [comment]
![image](https://github.com/user-attachments/assets/cc03e65a-65b8-48d3-9aeb-ff569280bea9)
